packages/backendv2/src/controllers/dashboard/index.ts

```diff
@@ -27,6 +27,37 @@ const dashboard = new Router<CustomState, CustomContext>({
     const courseId = ctx.params.courseId
     ctx.body = await Course.getFlattenedById(courseId)
   })
+  .post("/courses/:courseId/duplicate-course", async ctx => {
+    const oldCourseId = ctx.params.courseId
+    const token = ctx.request.body.token
+    const name = ctx.request.body.name
+    const abbr = ctx.request.body.abbr
+    const languageId = ctx.request.body.lang
+    if (!validToken(token)) {
+      ctx.body = "invalid token"
+    } else {
+      ctx.response.set("Content-Type", "text/csv")
+      ctx.response.attachment("quiz-transition.csv")
+      ctx.body = await Course.duplicateCourse(
+        oldCourseId,
+        name,
+        abbr,
+        languageId,
+      )
+    }
+  })
+  .post("/courses/download-correspondance-file", async ctx => {
+    const token = ctx.request.body.token
+    const oldCourseId = ctx.request.body.oldCourseId
+    const courseId = ctx.request.body.courseId
+    if (!validToken(token)) {
+      ctx.body = "invalid token"
+    } else {
+      ctx.response.set("Content-Type", "text/csv")
```

Comment:
It's not really csv since it's space separated. Maybe `text/plain` would work here better.